### PR TITLE
Max/lin 2305 bump revenuecat

### PR DESCRIPTION
**Pull request by myeyesareblind**
Bumping RC from 4.38.1 to 5.3.0-Snapshot

### CODE REVIEW

**Comment by myeyesareblind** [comment]
There were a few cases where I had to manually resolve (see few latest commit), otherwise no big surprises/tricky conflicts.
I've checked that UI looks the same & I was able to make a purchase in Curve

**Comment by myeyesareblind** [comment]
I think last time to review this I diffed this branch with 'main', to see what we have changed.

File: RevenueCatUI/Data/TemplateViewConfiguration.swift [hunk 1]
```diff
@@ -75,9 +75,16 @@ extension TemplateViewConfiguration {
...
+                return package.content.introductoryOfferDaysDuration
+            case .multiple(let mutliPackage):
+                return mutliPackage.all.first(where: { $0.content.introductoryOfferDaysDuration != nil })?.content.introductoryOfferDaysDuration
+            case let .multiTier(_, allMap, _):
+                for (_, multiPackage) in allMap {
+                    if let value = PackageConfiguration.multiple(multiPackage).introductoryOfferDaysDuration {
+                        return value
+                    }
+                }
+                return nil
```
Comment (g-laures) [inline RevenueCatUI/Data/TemplateViewConfiguration.swift]: Could be written this way:
```swift
var introductoryOfferDaysDuration: Int? {
    switch self {
    case .single(let package):
        package.content.introductoryOfferDaysDuration
    case .multiple(let multiPackage):
        multiPackage.introductoryOfferDaysDuration
    case let .multiTier(_, allMap, _):
        allMap.values.compactMap { $0.introductoryOfferDaysDuration }.first
    }
}
```
With the following extension being added:
```swift
@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
extension TemplateViewConfiguration.PackageConfiguration.MultiPackage {
    var introductoryOfferDaysDuration: Int? {
        all.first(where: { $0.content.introductoryOfferDaysDuration != nil })?.content.introductoryOfferDaysDuration
    }
}
```

**Review by g-laures — COMMENTED** [review]
I did not noticed any issue, only this code that could be updated

File: RevenueCatUI/Data/TemplateViewConfiguration.swift (hunk 1 above)
Comment (myeyesareblind) [reply RevenueCatUI/Data/TemplateViewConfiguration.swift]: updated 👍

**Review by g-laures — APPROVED** [review]

**Pull request merged**